AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/Support/MutationEventExtensionsTests.swift

```diff
@@ -88,69 +123,115 @@ class MutationEventExtensionsTest: BaseDataStoreTests {
+                XCTAssertEqual(last, pendingDeleteMutationEvent)
                 queryAfterUpdatingVersionExpectation.fulfill()
             }
         }
         wait(for: [queryAfterUpdatingVersionExpectation], timeout: 1)
     }
 
-    /// - Given: A create mutationSync with a given version
-    /// - When: Mutation event table contains update and delete mutation events added in order with `nil` version,
-    ///        having same model id
-    /// - Then: Update mutation event should be updated with version of create mutationSync
-    func testQueryAfterUpdatePendingMutationEventVersionGivenMultipleMutationEvents() {
+    /// - Given: A pending mutation events queue with event containing version 2, a sent mutation event model
+    ///         that matches the received mutation sync model having version 2. The received mutation sync has
+    ///         version 1.
+    /// - When: The sent model matches the received model and the first pending mutation event version is 2.
+    /// - Then: The first pending mutation event version should NOT be updated.
+    func testSentModelVersionNewerThanResponseVersion_PendingEventNotReconciled() throws {
         let modelId = UUID().uuidString
-        let post = AnyModel(Post(id: modelId, title: "title", content: "content", createdAt: .now()))
-        let updateMutationEvent = MutationEvent(id: UUID().uuidString,
-                                                        modelId: modelId,
-                                                        modelName: Post.modelName,
-                                                        json: "",
-                                                        mutationType: .update,
-                                                        version: nil)
-        let deleteMutationEvent = MutationEvent(id: UUID().uuidString,
-                                                        modelId: modelId,
-                                                        modelName: Post.modelName,
-                                                        json: "",
-                                                        mutationType: .delete,
-                                                        version: nil)
-        let metadata = MutationSyncMetadata(id: modelId,
-                                            deleted: false,
-                                            lastChangedAt: Int(Date().timeIntervalSince1970),
-                                            version: 1)
+        let post1 = Post(id: modelId, title: "title1", content: "content1", createdAt: .now())
+        let post2 = Post(id: modelId, title: "title2", content: "content2", createdAt: .now())
+        let requestMutationEvent =
+            try createMutationEvent(model: post1, mutationType: .create, version: 2, inProcess: true)
+        let pendingMutationEvent = try createMutationEvent(model: post2, mutationType: .update, version: 2)
```

Comment:
nit: remove mutationType parameter if it doesn't matter

Reply:
type is needed for `testSentModelWithNilVersion_SecondPendingEventNotReconciled` where we have `update` and `delete` events in queue and only `update` event is modified. Will let it be as the tests model the different scenarios like "Save, then update" etc.